analysis-module/src/analysis_cproc.py

```diff
@@ -0,0 +1,146 @@
+# Python Imports
+import re
+import sys
+import json
+from datetime import datetime
+
+# Project Imports
+from src import vars
+from src import analysis_logger
+
+def extract_node_id(s: str) -> str:
+    pattern = r"node-(\d+)\.toml"
+    match = re.search(pattern, s)
+    if match:
+        return f"node_{match.group(1)}"
+    else:
+        return None
+
+
+def load_process_level_metrics(metrics_file_path: str):
```

Comment:
I see that you are using Typing! Great, this is something I also wanted to use. If you all agree, we can start using it from now on.